noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -151,6 +152,33 @@ contract NFT {
         let to_note_slot = storage.private_nfts.at(to).storage_slot;
         let hiding_point = NFTNote::hiding_point().new(to_npk_m_hash, note_randomness, to_note_slot);
 
+        // ####### HACKY PARTIAL NOTE EMISSION CODE (to be replaced in a followup PR with proper macro support) #######
+        // TODO(#8769): The following is a temporary partial notes delivery solution that should be eventually replaced.
+        let to_public_keys = get_public_keys(to);
+        let hacky_nft_note = crate::hacky_partial_note::NFTNote {
+            npk_m_hash: to_npk_m_hash,
+            randomness: note_randomness,
+            header: dep::aztec::prelude::NoteHeader::empty()
+        };
+        // `from` cannot receive any logs here! so we encrypt outgoing to `to`
```

Comment:
Why is this?

Reply:
Clarified in https://github.com/AztecProtocol/aztec-packages/pull/8725/commits/1ec2e5d3e6ec16079eb13eb164535aa48f93dfb5

It's because it would reveal for whom the note was prepared.